gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -414,6 +443,18 @@ public boolean isAncestor(IGitCoreCommit presumedAncestor, IGitCoreCommit presum
     });
   }
 
+  @Override
+  public GitCoreRepositoryState deriveRepositoryState() {
+    return Match(jgitRepo.getRepositoryState()).of(
```

Comment:
Nit: maybe spotless formatting can be somehow turned off for a code region... it'd feel much better if it was consistently sth like
```java
        Case($(isIn(...)),
            ...),
```